### PR TITLE
One variable dialogue improvements

### PR DESCRIPTION
**Pull request by Ivanluv**
@africanmathsinitiative/developers  This is from PR #5861

### CODE REVIEW

**Comment by lloyddewit** [comment]
@Ivanluv Thank you for taking over PR #5861. I understand that:
- This PR completely replaces PR #5861 and includes all of its commits. Therefore when this PR is merged, PR #5861will also merge automatically.
- Your work on this PR is ongoing and is not yet ready for review


Please let me know if I misunderstood anything, thanks

**Comment by Ivanluv** [comment]
@lloyddewit  both  points are correct

**Comment by Ivanluv** [comment]
@rdstern  could you look at this again

**Review by rdstern — CHANGES_REQUESTED** [review]
That's looking good with the extra control.
a) It doesn't "remember" the level used previously.
b)When it isn't a factor variable the control isn't there.    Could the selector have anything except character.  Then when it is factor you can opt to make it numeric.  Could you also, when it is not a factor, and you are using binomial or proportion, it would have the equivalent control "Convert to Factor", and it would then do an as.factor.  
c) The selector does not allow character variables

**Review by rdstern — CHANGES_REQUESTED** [review]
This is now getting there.  There are 2 packages to add, namely DEscTools and MKinfer.
The other minor point is on the Z test, where Ok should be disabled if the sd field is empty.
Then I need @dannyparsons to help me on how to cope with variables that are not numeric.  I wonder what can (should) be coped with.

**Review by rdstern — APPROVED** [review]
@Ivanluv , please check that the new packages are included in the list to be added.

**Comment by Ivanluv** [comment]
@rdstern  I have added the package to that list

File: instat/dlgOneVarFitModel.vb [hunk 1]
```diff
@@ -14,400 +14,574 @@
...
-    Public clsRplotFunction, clsRplotPPComp, clsRplotCdfcomp, clsRplotQqComp, clsRplotDenscomp As RFunction
-    Public clsFunctionOperator, clsFactorOperator As New ROperator
+    Private clsROneVarFitModel, clsRLogLikFunction, clsRfitdist, clsRConvertVector, clsRConvertInteger, clsRConvertNumeric, clsNaExclude As New RFunction
+    Private clsRplotFunction, clsRplotPPComp, clsRplotCdfcomp, clsRplotQqComp, clsRplotDenscomp As New RFunction
+    Private clsBionomialFunction, clsProportionFunction, clsSignTestFunction, clsTtestFunction, clsWilcoxonFunction, clsZTestFunction, clsBartelFunction, clsBrFunction, clsRunsFunction, clsSenFunction, clsSerialCorrFunction, clsSnhFunction, clsAdFunction, clsCvmFunction, clsLillieFunction, clsPearsonFunction, clsSfFunction As New RFunction
+    Private clsMeanCIFunction, clsMedianCIFunction, clsNormCIFunction, clsQuantileCIFunction, clsSdCIFunction, clsVarCIFunction As New RFunction
+    Private clsGetFactorLevel As New RFunction
```
Comment (lloyddewit) [inline instat/dlgOneVarFitModel.vb:25]: Cahange all names to end in `Function` (e.g. rename `clsRConvertVector` to `clsRConvertVectorFunction`).
Also, some names need the capitalisation fixed (e.g. rename `clsRfitdist` to `clsRFitDistFunction`).

File: instat/dlgOneVarFitModel.vb [hunk 2]
```diff
@@ -14,400 +14,574 @@
...
         If ucrSaveModel.IsComplete() AndAlso Not ucrReceiverVariable.IsEmpty AndAlso ucrDistributionChoice.ucrInputDistributions.cboInput.SelectedItem <> "" Then
-            ucrBase.OKEnabled(True)
+            If ucrInputTxtSd.GetText = "" AndAlso rdoTest.Checked AndAlso ucrInputComboTests.GetText = "Z" Then
+                ucrBase.OKEnabled(False)
+            Else
+                ucrBase.OKEnabled(True)
+            End If
         Else
-            ucrBase.OKEnabled(False)
+                ucrBase.OKEnabled(False)
```
Comment (lloyddewit) [inline instat/dlgOneVarFitModel.vb:575]:
```suggestion
            ucrBase.OKEnabled(False)
```

File: instat/dlgOneVarFitModel.vb [hunk 3]
```diff
@@ -543,191 +615,321 @@ Public Class dlgOneVarFitModel
...
+        If rdoGeneralCase.Checked Then
+            ucrSaveModel.SetCheckBoxText("Save Model")
+            ucrSaveModel.SetPrefix("dist")
+        ElseIf rdoTest.Checked Then
+            ucrSaveModel.SetCheckBoxText("Save Test")
+            ucrSaveModel.SetPrefix("test")
+        ElseIf rdoEstimate.Checked Then
+            ucrSaveModel.SetCheckBoxText("Save Estimate")
+            ucrSaveModel.SetPrefix("ci")
+        End If
```
Comment (lloyddewit) [inline instat/dlgOneVarFitModel.vb:642]: add `else` that raises developer error?

File: instat/dlgOneVarFitModel.vb [hunk 4]
```diff
@@ -543,191 +615,321 @@ Public Class dlgOneVarFitModel
...
+                    Case "Bartel"
+                        tttests.SetToolTip(ucrInputComboTests.cboInput, "Bartels Rank Test of Randomness")
+                        clsBartelFunction.AddParameter("x", clsRFunctionParameter:=ucrReceiverVariable.GetParameter().clsArgumentCodeStructure, iPosition:=0)
+                    Case "runs"
+                        clsRunsFunction.AddParameter("x", clsRFunctionParameter:=ucrReceiverVariable.GetParameter().clsArgumentCodeStructure, iPosition:=0)
+                    Case "br"
+                        tttests.SetToolTip(ucrInputComboTests.cboInput, "Buishand Range Test for Change-Point Detection")
+                        clsBrFunction.AddParameter("x", clsRFunctionParameter:=ucrReceiverVariable.GetParameter().clsArgumentCodeStructure, iPosition:=0)
+                    Case "Sen"
+                        tttests.SetToolTip(ucrInputComboTests.cboInput, "Sen 's slope for linear rate of change")
```
Comment (lloyddewit) [inline instat/dlgOneVarFitModel.vb:756]:
```suggestion
                        tttests.SetToolTip(ucrInputComboTests.cboInput, "Sen's slope for linear rate of change")
```

File: instat/dlgOneVarFitModel.vb [hunk 5]
```diff
@@ -543,191 +615,321 @@ Public Class dlgOneVarFitModel
...
+                        clsSenFunction.AddParameter("x", clsRFunctionParameter:=ucrReceiverVariable.GetParameter().clsArgumentCodeStructure, iPosition:=0)
+                    Case "serial corr"
+                        clsSerialCorrFunction.AddParameter("x", clsRFunctionParameter:=clsRConvertVector, iPosition:=0)
+
```
Comment (lloyddewit) [inline instat/dlgOneVarFitModel.vb:760]:
```suggestion
```

File: instat/dlgOneVarFitModel.vb [hunk 6]
```diff
@@ -543,191 +615,321 @@ Public Class dlgOneVarFitModel
...
+                        clsLillieFunction.AddParameter("x", clsRFunctionParameter:=clsRConvertNumeric, iPosition:=0)
+                    Case "pearson"
+                        clsPearsonFunction.AddParameter("x", clsRFunctionParameter:=clsRConvertNumeric, iPosition:=0)
+                    Case "sf"
+                        clsSfFunction.AddParameter("x", clsRFunctionParameter:=clsRConvertNumeric, iPosition:=0)
+                    Case "ad"
+                        clsAdFunction.AddParameter("x", clsRFunctionParameter:=clsRConvertNumeric, iPosition:=0)
+                End Select
+            Else
+                Select Case ucrInputComboTests.GetText()
```
Comment (lloyddewit) [inline instat/dlgOneVarFitModel.vb:738]: Should all of these cases have tool tips? or just some?

File: instat/dlgOneVarFitModel.vb [hunk 7]
```diff
@@ -543,191 +615,321 @@ Public Class dlgOneVarFitModel
...
+                Case "normal"
+                    ucrBase.clsRsyntax.SetBaseRFunction(clsNormCIFunction)
+                Case "quantile"
+                    ucrBase.clsRsyntax.SetBaseRFunction(clsQuantileCIFunction)
+                Case "sd"
+                    ucrBase.clsRsyntax.SetBaseRFunction(clsSdCIFunction)
+                Case "variance"
+                    ucrBase.clsRsyntax.SetBaseRFunction(clsVarCIFunction)
+            End Select
+        End If
```
Comment (lloyddewit) [inline instat/dlgOneVarFitModel.vb:886]: add `else` with developer error?

File: instat/dlgOneVarFitModel.vb [hunk 8]
```diff
@@ -543,191 +615,321 @@ Public Class dlgOneVarFitModel
...
     End Sub
 
-    Private Sub AllControl_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrSaveModel.ControlContentsChanged, ucrReceiverVariable.ControlContentsChanged
-        TestOKEnabled()
+    ''' <summary> 
+    '''Lists the factor levels of the selected factor or logical column into the success combobox
+    ''' </summary>
+
```
Comment (lloyddewit) [inline instat/dlgOneVarFitModel.vb:892]:
```suggestion
```

File: instat/dlgOneVarFitModel.vb [hunk 9]
```diff
@@ -543,191 +615,321 @@ Public Class dlgOneVarFitModel
...
+    Private Sub AddFactorLevels()
+        If bRCodeSet Then
+            Dim chrCurrentFactorLevels As CharacterVector
+            Dim lstFactor As List(Of String) = New List(Of String)()
+            If Not ucrReceiverVariable.IsEmpty AndAlso rdoTest.Checked AndAlso (ucrReceiverVariable.strCurrDataType.ToLower = "logical" Or ucrReceiverVariable.strCurrDataType.ToLower = "factor") AndAlso (ucrInputComboTests.GetText() = "binomial" Or ucrInputComboTests.GetText() = "proportion") Then
```
Comment (lloyddewit) [inline instat/dlgOneVarFitModel.vb:897]: Convert this condition to multiple lines so that no single line > col 100

**Review by lloyddewit — CHANGES_REQUESTED** [review]
@Ivanluv Looks good! I just have some relatively trivial comments

File: instat/dlgOneVarFitModel.vb (hunk 1 above)
Comment (Ivanluv) [reply instat/dlgOneVarFitModel.vb:25]: Done

File: instat/dlgOneVarFitModel.vb (hunk 3 above)
Comment (Ivanluv) [reply instat/dlgOneVarFitModel.vb:642]: How can I do that?

File: instat/dlgOneVarFitModel.vb (hunk 6 above)
Comment (Ivanluv) [reply instat/dlgOneVarFitModel.vb:738]: Some

File: instat/dlgOneVarFitModel.vb (hunk 7 above)
Comment (Ivanluv) [reply instat/dlgOneVarFitModel.vb:886]: Any suggestion on how to do that

File: instat/dlgOneVarFitModel.vb (hunk 3 above)
Comment (lloyddewit) [reply instat/dlgOneVarFitModel.vb:642]: I moved this comment to PR #6150

File: instat/dlgOneVarFitModel.vb (hunk 6 above)
Comment (lloyddewit) [reply instat/dlgOneVarFitModel.vb:738]: ok

File: instat/dlgOneVarFitModel.vb (hunk 7 above)
Comment (lloyddewit) [reply instat/dlgOneVarFitModel.vb:886]: I added this comment to PR #6150

**Pull request closed without merging**